share/spice/js_keycodes/js_keycodes.js

```diff
@@ -0,0 +1,155 @@
+            data = {};
+
+        // if the query was something like "keycode tab"
+        // display the result for that
+        // otherwise we render the keycode tester
+        if (key) {
+            data.result = getKeycodeByKey(key);
+        }
+
+        // grab all the codes for the keycodes table
+        // manually add the index as this version of handlebars
+        // doesn't support @index inside #each
+        data.codes = $.map(keycodeLookup, function (keycode, index) {
+            return {
+                key: keycode.key,
+                code: keycode.code,
+                index: index
+            };
+        });
+
+        Spice.registerHelper({
+            // output three keycodes per row
+            tr: function (options) {
+                var markup = "";
+                if (this.index % 3 === 0) {
+                    markup += "<tr>";
+                }
+
+                markup += options.fn(this);
+
+                if (this.index % 3 === 2) {
+                    markup += "</tr>";
+                }
+
+                return markup;
+            },
+            // add separator between keycodes
+            separator: function () {
+                if (this.index % 3 !== 2) {
+                    return "class='with-separator'";
+                }
+            }
+        });
+
+        Spice.add({
+            id: "js_keycodes",
+            name: "Keycodes",
+            data: data,
+            meta: {
+                sourceName: "MDN",
+                sourceUrl: "https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode"
+            },
+            templates: {
+                group: 'base',
+                options: {
+                    content: Spice.js_keycodes.content,
+                    moreAt: true
+                }
+            },
+            onShow: onShow
+        });
+
+        function onShow() {
+            var $spice = Spice.getDOM("js_keycodes"),
+                $table = $spice.find("table.all-keycodes"),
+                $input = $spice.find("input.keycode-input"),
+                $result = $spice.find("p.keycode-result");
+
+            // hide / show keycodes table
+            $(".zci--js_keycodes .show-keycodes").click(function (e) {
+                e.preventDefault();
+
+                $table.toggleClass("is-hidden");
+            });
+
+            // handle tester input
+            $input.keydown(function (e) {
+                var code = e.keyCode || e.which,
+                    keyCode = getKeycodeByCode(code),
+                    keyName;
+
+                e.preventDefault();
+
+                // in case keycode is unknown
+                if (!keyCode) {
+                    return;
+                }
+
+                keyName = keyCode.key;
+
+                $input.val(keyName);
+                $result.html("The keycode for <code>" + keyName + "</code> is: " + code);
+            });
+        }
+
+    };
```

Comment:
Please call `ddg_spice_js_keycodes` at the end like https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/share/spice/flash_version/flash_version.js#L34. :)